src/scripts/main.js

```diff
@@ -1,3 +1,282 @@
+
+  titleElement.textContent = title;
+
+  descriptionElement.textContent = description;
+  titleElement.classList.add('title');
+  messageElement.append(titleElement, descriptionElement);
+
+  messageElement.classList.add(type, 'notification');
+  messageElement.setAttribute('data-qa', 'notification');
+
+  document.body.append(messageElement);
+
+  setTimeout(() => {
+    messageElement.remove();
+  }, 3000);
+};
+
+const saveChanges = function(editInput, clicked, pereviousValue) {
+  const editedValue = editInput.value;
+
+  clicked.innerHTML = '';
+
+  if (editedValue === '') {
+    clicked.textContent = pereviousValue;
+  } else {
+    clicked.textContent = editedValue;
+  }
+};
+
+/// /// /// /// /// /// /// ///
+// TABLE SORTING
+/// /// /// /// /// /// /// ///
+
+const sorted = [0, 0, 0, 0, 0];
+
+const sortTable = function(e) {
+  const tableData = [...tableBody.children];
+  const clicked = e.target;
+  const type = clicked.textContent;
+  const index = [...clicked.closest('tr').children].indexOf(clicked);
+
+  tableData.sort((a, b) => {
+    const firstText = a.children[index].textContent;
+    const secondText = b.children[index].textContent;
+
+    if (sorted[index] === 0) {
+      switch (type) {
+        case 'Position':
+        case 'Name':
+        case 'Office':
+          return firstText.localeCompare(secondText);
+        case 'Age':
+          return firstText - secondText;
+        case 'Salary':
+          return getSalaryNumber(firstText) - getSalaryNumber(secondText);
+      };
+    } else {
+      switch (type) {
+        case 'Position':
+        case 'Name':
+        case 'Office':
+          return secondText.localeCompare(firstText);
+        case 'Age':
+          return secondText - firstText;
+        case 'Salary':
+          return getSalaryNumber(secondText) - getSalaryNumber(firstText);
+      }
+    }
+  });
+
+  sorted.forEach((_, i) => {
+    if (i === index) {
+      sorted[index] = sorted[index] === 0 ? 1 : 0;
+    } else {
+      sorted[i] = 0;
+    }
+  });
+
+  tableBody.innerHTML = '';
+  tableData.forEach(el => tableBody.append(el));
+};
+
+document.querySelector('thead').addEventListener('click', sortTable);
+
+/// /// /// /// /// /// /// ///
+// SELECTED ROW
+/// /// /// /// /// /// /// ///
+
+const makeRowSelected = function(e) {
+  const clicked = e.target.closest('tr');
+
+  if (!clicked) {
+    return;
+  }
+
+  [...document.querySelectorAll('tr')].forEach(
+    tr => tr.classList.remove('active')
+  );
+  clicked.classList.add('active');
+};
+
+tableBody.addEventListener('click', makeRowSelected);
+
+/// /// /// /// /// /// /// ///
+// FORM
+/// /// /// /// /// /// /// ///
+
+// 1) creating the form
+
+const formElement = document.createElement('form');
+
+formElement.classList.add('new-employee-form');
+
+const inputName = createInput('Name:', 'name');
+const inputPosition = createInput('Position:', 'position');
+
+// 2) Select Office
+
+const lableSelect = document.createElement('label');
```

Comment:
Here we have a lot of similar action. 

We can create an array of cities and then using a loop (or even better an array method), we can create all of elements, setting attributes etc

`const cities = ['Tokyo', 'Singapore' .... ] `